### ISSUE TITLE
A `to_slice` method for primitive types

### ISSUE THREAD

**Issue by comath** [capnproto/capnproto-rust#239]
I want to store points for a machine learning library in cap'n proto, and would like to access the dense data as a slice of floats. Cap'n proto stores everything as little endian, and I believe it should be possible on little endian systems to safely return the data as slices. Is it possible and would it be welcome? 

Something like this:
```
#[cfg(target_endian = "little")]
pub trait PrimitiveListToSlice<T: PrimitiveElement> {
    fn to_slice(&self) ->&[T];
}
```
I believe it only reads on contiguous data, so we don't need to worry about the message being split over a few buffers. If we do, we can return an `Option`, or a `Vec` of slices, but that's way more complicated than what I think we need to do to add this feature.

**Comment by dwrensha** [comment]
We do have something along these lines in `capnp::raw::get_list_bytes()`:

https://github.com/capnproto/capnproto-rust/blob/9cf00528a6a92ba57c835fd467f5217d84e5707b/capnp/src/raw.rs#L54-L59

So I suppose the only missing piece would be the conversion from `&[u8]` to `&[f32]`.
My initial feeling is that you should write your own little function do that `transmute` on your end, but I could maybe be convinced that something like that belongs within the `capnp` crate.

**Comment by comath** [comment]
I'm working on a evaluation library, and have added the recommended transmute in there. The transmute needs a check for little-endianness and a use of unsafe. It gets a little messy, and I'm going to need to do this a lot of places.

[Flatbuffers](https://docs.rs/flatbuffers/2.0.0/flatbuffers/struct.Vector.html#method.safe_slice) have a 'to_slice' method :grin:.

**Comment by dwrensha** [comment]
closed by #240.
renamed to `as_slice()` in f77e49986df561b82de8219affda75b043f1afb9
